tests/components/generic_thermostat/test_climate.py

```diff
@@ -159,131 +171,6 @@ async def test_heater_switch(hass, setup_comp_1):
     assert STATE_ON == hass.states.get(heater_switch).state
 
 
-def _setup_sensor(hass, temp):
```

Comment:
Just a sidenote as I don't know how coverage is being measured in this repository:

You might focus on implementing your feature. If you are changing the whole testing behavior instead of just adding tests for your feature, it is hard to verify if your change has BC breaks or not as the diff contains too many unrelated changes.

So instead of reworking the whole tests, you might just want to add tests for your feature instead.
Refactoring the tests should probably be done in a different PR as shipping features is more valuable than refactoring. A refactoring on the tests can be merged somewhen in the future but having this feature available suits more users.

Reply:
This is true, but in the other hand to be sure that the new feature work well, and still backward compatible, I've to duplicate a lot of test, to check the integration work with `away_temp` and with `presets`
Keeping all test together make the file very huge. I can roll back the file splitting, and make another PR, to split the tests for maintainability.
In a future, when the `away_temp` will be removed, the file test_climate_away_temp_config.py can be trashed.

Reply:
Thanks for the feedback. I totally understand the "why" but at the end its not me who has to merge this.

as homeassistant is huge and thus the developers focus on their own roadmaps, its more likely they merge "easy picks" to new versions as they dont change too many things. 

If the PR takes too much time to review as it changes too many LoC, it just wont make it to the next version. 
I can tell that as an oss maintainer.